File: backend/src/main/java/com/foodbridge/volunteer/service/VolunteerService.java
package com.foodbridge.volunteer.service;

import java.util.List;

import com.foodbridge.volunteer.dto.UpdateAvailabilityRequest;
import com.foodbridge.volunteer.dto.VolunteerDashboardResponse;
import com.foodbridge.volunteer.dto.VolunteerDeliveryResponse;
import com.foodbridge.volunteer.dto.VolunteerProfileResponse;
import com.foodbridge.profile.dto.request.VolunteerProfileRequest;

public interface VolunteerService {

    // Profile

    VolunteerProfileResponse getProfile();

    String updateProfile(VolunteerProfileRequest request);

    // Dashboard
    VolunteerDashboardResponse getDashboard();

    // Deliveries
    List<VolunteerDeliveryResponse> getAssignedDeliveries();

    VolunteerDeliveryResponse getDeliveryDetails(Long deliveryId);

    String markPickup(Long deliveryId);

    String markInTransit(Long deliveryId);

    String markDelivered(Long deliveryId);

    // Availability
    String updateAvailability(UpdateAvailabilityRequest request);

    // History
    List<VolunteerDeliveryResponse> getDeliveryHistory();
}

File: frontend/src/services/volunteerService.js
import api from "./api";

const volunteerService = {

    getProfile: async () => {
        const response = await api.get("/volunteer/profile");
        return response.data;
    },

    updateProfile: async (data) => {
        const response = await api.put("/volunteer/profile", data);
        return response.data;
    },

    getDashboard: async () => {
        const response = await api.get("/volunteer/dashboard");
        return response.data;
    },

    getDeliveries: async () => {
        const response = await api.get("/volunteer/deliveries");
        return response.data;
    },

    getDeliveryDetails: async (deliveryId) => {
        const response = await api.get(
            `/volunteer/deliveries/${deliveryId}`
        );
        return response.data;
    },

    updateAvailability: async (available) => {
        const response = await api.put(
            "/volunteer/availability",
            { available }
        );
        return response.data;
    },

    markPickup: async (deliveryId) => {
        const response = await api.put(
            `/volunteer/deliveries/${deliveryId}/pickup`
        );
        return response.data;
    },

    markInTransit: async (deliveryId) => {
        const response = await api.put(
            `/volunteer/deliveries/${deliveryId}/transit`
        );
        return response.data;
    },

    markDelivered: async (deliveryId) => {
        const response = await api.put(
            `/volunteer/deliveries/${deliveryId}/delivered`
        );
        return response.data;
    },

    getDeliveryHistory: async () => {
        const response = await api.get("/volunteer/history");
        return response.data;
    },
};

export default volunteerService;